rxandroidble/src/main/java/com/polidea/rxandroidble/internal/scan/ExcessiveScanCheckerApi24.java

```diff
@@ -0,0 +1,45 @@
+package com.polidea.rxandroidble.internal.scan;
+
+
+import android.support.annotation.Nullable;
+import java.util.Date;
+
+public class ExcessiveScanCheckerApi24 implements ExcessiveScanChecker {
+
+    /*
+     * default values taken from
+     * https://android.googlesource.com/platform/packages/apps/Bluetooth/+/android-7.0.0_r1/src/com/android/bluetooth/gatt/AppScanStats.java
+     */
+    private static final int SCANS_LENGTH = 5;
+
+    private static final long EXCESSIVE_SCANNING_PERIOD = 30 * 1000;
+
+    private final long[] previousChecks = new long[SCANS_LENGTH];
+
+    @Nullable
+    @Override
+    public Date suggestDateToRetry() {
```

Comment:
It's not obvious that the method returning a suggestion is also modifying the state of the checker. I think that this method should idempotent and there should be a method to alter the state.